internal/registry/registry.go

```diff
@@ -224,22 +227,31 @@ func (r *registry) getLayersDigestsFromManifestStream(manifestStream []byte) ([]
 }
 
 func (r *registry) GetHeaderDataFromLayer(layer v1.Layer, headerName string) ([]byte, error) {
-
-	targz, err := layer.Compressed()
+	reader, readerCloser, err := r.getHeaderReaderFromLayer(layer, headerName)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get targz from layer: %w", err)
+		return nil, fmt.Errorf("failed to get reader for the header %s in the layer: %v", headerName, err)
 	}
-	// err ignored because we're only reading
-	defer targz.Close()
+	defer readerCloser.Close()
+	buff, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read header %s data: %v", headerName, err)
+	}
+	return buff, nil
+}
 
-	gr, err := gzip.NewReader(targz)
+func (r *registry) getHeaderReaderFromLayer(layer v1.Layer, headerName string) (io.Reader, io.ReadCloser, error) {
+	layerreader, err := layer.Uncompressed()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create reader from targz: %w", err)
+		return nil, nil, fmt.Errorf("failed to get layerreader from layer: %w", err)
 	}
 	// err ignored because we're only reading
-	defer gr.Close()
+	defer func() {
+		if err != nil {
```

Comment:
Seems like this func is enclosing `err`. It may change later in this function to contain the value of another action error instead of the intended one.

Reply:
As discussed, this is the intention: we want to close ReadCloser in case of any error in the function